src/battle_ai_switch_items.c

```diff
@@ -990,8 +990,7 @@ u8 GetMostSuitableMonToSwitchInto(void)
         return bestMonId;
 
     // If ace mon is the last available Pokemon and U-Turn/Volt Switch was used - switch to the mon.
-    if (aceMonId != PARTY_SIZE
-        && (gBattleMoves[gLastUsedMove].effect == EFFECT_HIT_ESCAPE || gBattleMoves[gLastUsedMove].effect == EFFECT_PARTING_SHOT))
+    if (aceMonId != PARTY_SIZE)
```

Comment:
This is the part that I think is truly fixing the bug.

Reply:
Yep, this is the actual fix. I am honestly not sure why I added the check above yesterday.